internal/dns/dns.go

```diff
@@ -44,67 +45,98 @@ func NewDNSHandler(baseDNSDomains map[string]string, log logrus.FieldLogger) DNS
 	}
 }
 
-func (h *handler) ChangeDNSRecordSets(ctx context.Context, cluster *common.Cluster, delete bool) error {
+func (h *handler) CreateDNSRecordSets(ctx context.Context, cluster *common.Cluster) error {
 	log := logutil.FromContext(ctx, h.log)
+	if err := h.updateDNSRecordSet(log, cluster, h.createDNSRecord); err != nil {
+		return err
+	}
+	log.Infof("Successfully created DNS records for base domain: %s", cluster.BaseDNSDomain)
+	return nil
+}
+
+func (h *handler) DeleteDNSRecordSets(ctx context.Context, cluster *common.Cluster) error {
+	log := logutil.FromContext(ctx, h.log)
+	if err := h.updateDNSRecordSet(log, cluster, h.deleteDNSRecord); err != nil {
+		return err
+	}
+	log.Infof("Successfully deleted DNS records for base domain: %s", cluster.BaseDNSDomain)
+	return nil
+}
 
+func (h *handler) updateDNSRecordSet(log logrus.FieldLogger, cluster *common.Cluster, updateRecordFunc func(logrus.FieldLogger, *DNSDomain, string, string) error) error {
```

Comment:
Maybe worth soring the context/log in the handler so we won't have to pass the log to all internal functions here. Could simplify the signatures.

Reply:
I'm not sure. IIUC, the handler is reused between requests, while the context (and log created from it) is per request.

Reply:
Yeah I've meant storing it only when calling the public functions (CreateDNSRecordSets/DeleteDNSRecordSets) then reuse it in the helpers. Just if worth the effort.